app/models/circle.rb

```diff
@@ -78,8 +139,28 @@ def allocated_stem(user_id)
     Track.find(self.allocation_hash["#{user_id}"])
   end
 
+  def pending_remix_from(stem)
+    user_id = self.allocation_hash.key(stem.id).first
+    User.find(user_id)
+  end
+
   def stem_from(user)
     return false unless self.has_member(user)
     Submission.where(circle_id: self.id).detect { |submission| submission.original.creator == user}
   end
+
+  def mixup
+    return false if self.members.count < 3
+    stems = self.submissions.map { |submission| submission.original.id }
+    remixers = self.members.map { |member| member.id }
```

Comment:
Redundant `self` detected.<br>Pass `&:id` as an argument to `map` instead of a block.